### PR TITLE
Update recipe for v1.0.0

### PR DESCRIPTION
**Pull request by OriolAbril**
Updated versions and requirements to match those in https://github.com/executablebooks/MyST-NB/blob/af2d1970306dfc5056a3e7c49cc0351bf327ef94/pyproject.toml

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by OriolAbril** [comment]
@conda-forge-admin, please rerender

**Comment by OriolAbril** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 
 package:
-  name: myst-nb
+  name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/m/myst-nb/myst-nb-{{ version }}.tar.gz
-  sha256: 0f61386515fab07c73646adca97fff2f69f41e90d313a260217c5bbe419d858b
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name.replace('-', '_') }}-{{ version }}.tar.gz
```
Comment (bollwyvl) [inline recipe/meta.yaml:9]: this is why I don't bother with the `name` junk these days... probably just put it  back to being explict in all places.

Comment (OriolAbril) [reply recipe/meta.yaml:9]: sounds good

Comment (OriolAbril) [reply recipe/meta.yaml:9]:
```suggestion
  url: https://pypi.io/packages/source/m/myst-nb/myst_nb-{{ version }}.tar.gz
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,12 +1,13 @@
-{% set version = "0.17.2" %}
+{% set name = "myst-nb" %}
+{% set version = "1.0.0" %}
 
 package:
-  name: myst-nb
+  name: {{ name|lower }}
```
Comment (bollwyvl) [inline recipe/meta.yaml:5]: also this

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/m/myst-nb/myst-nb-{{ version }}.tar.gz
-  sha256: 0f61386515fab07c73646adca97fff2f69f41e90d313a260217c5bbe419d858b
+  url: https://pypi.io/packages/source/m/myst-nb/myst_nb-{{ version }}.tar.gz
+  sha256: 9077e42a1c6b441ea55078506f83555dda5d6c816ef4930841d71d239e3e0c5e
 
 build:
   number: 0
```
Comment (bollwyvl) [inline recipe/meta.yaml:12]: let's also add the belt-and-suspenders `--no-build-isolation` to `pip install` to catch build system drift

**Review by bollwyvl — APPROVED** [review]
LGTM: will leave open for a bit so @conda-forge/jupyter-book and @conda-forge/myst-nb can weigh in

**Pull request merged**